daprblog/content/posts/2023/v1.12-release.md

```diff
@@ -0,0 +1,482 @@
+
+### State Management: Outbox Pattern for state stores (preview feature)
+
+The [outbox pattern](https://v1-12.docs.dapr.io/developing-applications/building-blocks/state-management/howto-outbox/) enables a single transaction across a state store and any message broker and is a powerful design pattern for sending notifications regarding changes in an application’s state.
+
+In this v1.12 release, the outbox pattern is enabled as a preview feature for any transactional state store.
+
+### Dapr Workflow in beta, with Java support
+
+The workflow runtime and APIs have been moved to [beta status](https://github.com/dapr/proposals/blob/7138b9c2835b91dfc3a16570b59882430a7940c8/guides/api-design.md#api-lifecycle-expectations) indicating the next stage of maturity towards stable. The [beta APIs](https://v1-12.docs.dapr.io/operations/support/alpha-beta-apis/#beta-apis) are the preferred choice when calling the Workflow APIs directly. [Note: the alpha APIs remain functional to prevent breaking changes]
+
+Additionally, [Java SDK](https://v1-12.docs.dapr.io/developing-applications/building-blocks/workflow/workflow-overview/#workflow-sdks) is now supported for authoring Workflows, which you can try through the [Java workflow quickstarts](https://v1-12.docs.dapr.io/developing-applications/building-blocks/workflow/workflow-overview/#try-out-workflows).
+
+### Metadata API improvements
+
+The [metadata API](https://v1-12.docs.dapr.io/reference/api/metadata_api/) on the [Dapr sidecar](https://v1-12.docs.dapr.io/concepts/dapr-services/sidecar/) now includes information on the [app connection details](https://v1-12.docs.dapr.io/reference/api/metadata_api/#app-connection-details) including the app port, protocol, host, max concurrency, along with health check details as seen from the Dapr sidecar.
+
+### Improvement to Dapr actors and Placement APIs
+
+In this v1.12 release we continued to make improvements to Dapr actors, including bug fixes and performance improvements to Actor Reminders.
+
+Some notable changes include:
+
+- A new [placement API](https://v1-12.docs.dapr.io/reference/api/placement_api/) on the [Placement service](https://v1-12.docs.dapr.io/concepts/dapr-services/placement/#placement-tables) enables you to examine the placement tables to determine which types of actors are deployed and where they are active. This gives you more insight to actors at runtime and aids in debugging actor issues.
+- API calls to Actor (and Workflow) endpoints now block while the Dapr actor runtime is being initialized, so applications don't need to guess anymore when the actor runtime is ready after Dapr has started.
+- We have made improvements to the performance of Actor Reminders, reducing reducing the need to perform multiple re-evaluations when more than one Dapr sidecar goes online and/or offline at the same time, and better handling locking to prevent race conditions.
```

Comment:
```suggestion
- We have made improvements to the performance of Actor Reminders, reducing the need to perform multiple re-evaluations when more than one Dapr sidecar goes online and/or offline at the same time, and better handling locking to prevent race conditions.
```